internal/providers/terraform/google/container_cluster.go

```diff
@@ -0,0 +1,107 @@
+package google
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/tidwall/gjson"
+
+	"github.com/shopspring/decimal"
+)
+
+func GetContainerClusterRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "google_container_cluster",
+		RFunc: NewContainerCluster,
+		Notes: []string{
+			"Sustained use discounts are applied to monthly costs, but not to hourly costs.",
+			"Costs associated with non-standard Linux images, such as Windows and RHEL are not supported.",
+			"Custom machine types are not supported.",
+		},
+	}
+}
+
+func NewContainerCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	description := "Regional Kubernetes Clusters"
+	region := d.Get("location").String()
+	if isZone(region) {
+		region = zoneToRegion(region)
+		description = "Zonal Kubernetes Clusters"
+	}
+
+	subResources := make([]*schema.Resource, 0)
+
+	if !d.Get("remove_default_node_pool").Bool() {
+		zones := int64(zoneCount(d.RawValues, ""))
+
+		countPerZone := int64(3)
+
+		if d.Get("initial_node_count").Type != gjson.Null {
```

Comment:
why not `d.Get("initial_node_count").Exists()`? to catch cases where it exists but is an empty string?

Reply:
`.Exists()` is a weird one. It only checks if the value is present in the JSON, which it is always is in this case (https://github.com/tidwall/gjson/issues/104). It doesn't check for blank values. Checking `.Type != gjson.Null` checks for non-existent and null values, which catches this case okay - this one is a number so don't get empty strings.

Reply:
We have other places that do `if d.Get("iops").Exists() && d.Get("iops").Type != gjson.Null` - just checking the `.Exists()` isn't needed?